src/components/Publication/Publication.js

```diff
@@ -75,22 +75,29 @@ class Publication extends Component {
       subDoc: '',
     };
 
-    this.arethusa = new ArethusaWrapper();
-    this.arethusaSubDocFun = this.arethusaSubDocFun.bind(this);
-  }
+    this.setSubdoc = this.setSubdoc.bind(this);
 
-  componentDidMount() {
+    this.arethusa = new ArethusaWrapper();
     // eslint-disable-next-line no-undef
-    window.arethusaSubDocFun = this.arethusaSubDocFun;
+    this.interval = window.setInterval(this.setSubdoc, 100);
   }
 
   componentWillUnmount() {
     // eslint-disable-next-line no-undef
-    window.arethusaSubDocFun = undefined;
+    window.clearInterval(this.interval);
```

Comment:
so is this the only time the timer is unset? When does it get called?

Reply:
In React `componentWillUnmount` is called immediately before the component is removed (unmounted). Usually API calls, listeners, and other async things are set up in a function called `componentDidMount` (which executes immediately after the component is rendered) and torn down in `componentWillUnmount`. While looking into this I realized that here I put the original `setInterval` in the constructor instead of `componentDidMount` so I will fix that!

Reply:
ok. but does that mean the timer continues the entire time the page is loaded?

Reply:
Yeah, the timer continues executing while the page is loaded. But the component itself is only re-rendered when the value of the `subDoc` is different. The way the code is written right now, the timer is also what causes the `subDoc` to change on the page when the user visits another chunk.

I could be wrong but I don't think this would be a performance issue with the timer only executing once every 100ms, even though it's not ideal.